recipe-client-addon/lib/NormandyApi.jsm

```diff
@@ -85,16 +115,20 @@ this.NormandyApi = {
    * Fetch metadata about this client determined by the server.
    * @return {object} Metadata specified by the server
    */
-  classifyClient() {
-    return this.get("classify_client/")
-      .then(response => response.json())
-      .then(clientData => {
-        clientData.request_time = new Date(clientData.request_time);
-        return clientData;
-      });
-  },
+  classifyClient: Task.async(function* () {
+    const classifyClientUrl = (yield this.apiIndex())["classify-client"];
+    const response = yield this.get(classifyClientUrl);
+    const clientData = yield response.json();
+    clientData.request_time = new Date(clientData.request_time);
+    return clientData;
+  }),
 
-  fetchAction(name) {
-    return this.get(`action/${name}/`).then(req => req.json());
-  },
+  fetchAction: Task.async(function* (name) {
+    let actionApiUrl = (yield this.apiIndex())["action-list"];
+    if (actionApiUrl.slice(-1) !== "/") {
```

Comment:
Here and elsewhere, you can use `str.endsWith()` instead of `.slice`, which is a bit easier to read.